src/main/java/com/lunark/lunark/notifications/service/NotificationService.java

```diff
@@ -63,13 +65,67 @@ public Notification createNotification(Review review)  {
         return this.create(notification);
     }
 
+
     private String createReviewNotificationMessage(Review review) {
         if (review.getType().equals(Review.ReviewType.HOST)) {
             return "New host review";
         }
         return "New property review for " + review.getProperty().getName();
     }
 
+    @Override
+    public Notification createNotification(Reservation reservation) {
+        String message = createReservationNotificationMessage(reservation);
+        Account account = determineAccountForNotification(reservation);
+        NotificationType notificationType = determineNotificationType(reservation);
+
+        Notification notification = new Notification(
+                message,
+                ZonedDateTime.now(clock),
+                notificationType,
+                account
+        );
+
+        return this.create(notification);
+    }
+
+    private Account determineAccountForNotification(Reservation reservation) {
+        if (reservation.getStatus().equals(ReservationStatus.ACCEPTED) ||
+                reservation.getStatus().equals(ReservationStatus.REJECTED)) {
+            return reservation.getGuest();
+        } else if (reservation.getStatus().equals(ReservationStatus.CANCELLED)) {
+            return reservation.getProperty().getHost();
+        }
+        throw new IllegalArgumentException("Invalid reservation status");
+    }
+
+    private NotificationType determineNotificationType(Reservation reservation) {
+        switch (reservation.getStatus()) {
+            case ACCEPTED:
+                return NotificationType.RESERVATION_ACCEPTED;
+            case REJECTED:
+                return NotificationType.RESERVATION_REJECTED;
+            case CANCELLED:
+                return NotificationType.RESERVATION_CANCELED;
+            default:
+                throw new IllegalArgumentException("Invalid reservation status");
+        }
+    }
+
+    private String createReservationNotificationMessage(Reservation reservation) {
+        switch (reservation.getStatus()) {
+            case ACCEPTED:
+                return "Reservation at has been accepted.";
+            case REJECTED:
+                return "Reservation at has been rejected.";
+            case CANCELLED:
+                return "Reservation at has been cancelled." ;
```

Comment:
These notifications messages probably won't go over the 255 character limit imposed by the database schema.
```suggestion
 case ACCEPTED:
                return "Reservation at " + reservation.getProperty().getName() + " has been accepted.";
            case REJECTED:
                return "Reservation at " + reservation.getProperty().getName() + " has been rejected.";
            case CANCELLED:
                return "Reservation at " + reservation.getProperty().getName() + " has been rejected by " + reservation.getGuest().getName() + " " + reservation.getGuest().getSurname();
```